eng/native/configuretools.cmake

```
@@ -1,3 +1,7 @@
+if(NOT CLR_CMAKE_CONFIGURE_PLATFORM_INCLUDED)
+  message(FATAL_ERROR "configuretools.cmake needs to be included after configureplatforms.cmake")
+endif()
```

Could we instead:

```suggestion
include(${CMAKE_CURRENT_LIST_DIR}/configureplatform.cmake)
```

and remove three occurrences of `include(${CLR_ENG_NATIVE_DIR}/configuretools.cmake)` under `:/src/`? This way we would only use `configureplatform.cmake` as an entrypoint cmake for eng/native (and rest of the files will become internal detail, which however may get refactored in the future).

---

or perhaps for all call sites, we can add an explicit `:/eng/native/entrypoint.cmake` enlisting:

```cmake
include(${CMAKE_CURRENT_LIST_DIR}/features.cmake)
include(${CMAKE_CURRENT_LIST_DIR}/configureplatform.cmake)
include(${CMAKE_CURRENT_LIST_DIR}/configuretools.cmake)
```

---

Yes I think this makes sense. I'd prefer if we do it in a separate PR though :)

---

I have addressed it as part of Android fixes https://github.com/dotnet/runtime/pull/32800.